### PR TITLE
Delete emails

### PR DESCRIPTION
**Pull request by seiferla**

### CODE REVIEW

**Review by jens1o — APPROVED** [review]
nice work! :)

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/organize/delete/DeleteEmailService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      // so there is nothing left to do
+
+      didFindEmail = true;
+      mailboxRepository.save(mailbox);
+      break;
+    }
+
+    if (!didFindEmail) {
+      throw new IllegalStateException("email not found in any mailbox");
+    }
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/organize/delete/DeleteEmailService.java:54]: https://github.com/DHBW-Mailflow/mailserver/pull/134#discussion_r1581418081
try adding another method here for easier understanding

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/DeleteEmailsCLIPrompt.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/DeleteEmailsCLIPrompt.java:1]: move this file to a usecases/delete

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/DeleteEmailCLIPrompt.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/DeleteEmailCLIPrompt.java:1]: move this file to a usecases/delete

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]
Just some small adjustments. I also mentioned one in #134 but due to copy paste, i had to mark this one here too

**Comment by Jonas-Karl** [comment]
Is it possible to avoid the duplicated code? Why do we need the DeleteEmailsCLIPrompt class, when the functionality is nearly the same as in MarkEmailsCLIPrompt? Maybe renaming the MarkEmailsInterface to OrganizeEmailsUseCase and renaming the method to `organize(Email email)` would avoid the duplicated code. But this is just a guess, I did not try it out

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/organize/delete/DeleteEmailService.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/organize/delete/DeleteEmailService.java:1]: move this into the parent folder

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/organize/delete/DeleteEmailsUseCase.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/organize/delete/DeleteEmailsUseCase.java:1]: move this into the parent folder

**Comment by Jonas-Karl** [comment]
The duplicated code is fine, because we will use this in our presentation

**Review by Jonas-Karl — APPROVED** [review]
I still need to test it in the IDE but the diff looks good

**Comment by Jonas-Karl** [comment]
Works fine! You could also filter out the DELETED mailbox, so that we can not deleted a deleted email again. Anyways its not a big issue because nothing crashes. I think this can be merged. Feel free to do it tomorrow :)

**Pull request merged**